### PR TITLE
Fixed Search Debounce Functionality in Org People Filters

### LINKED ISSUE #893: Debounce Functionality Doesn't Work in Org People Filters
**Describe the bug**
The debounce functionality in Organization People's filter doesn't work

**To Reproduce**

1. Run Talawa Admin
2. Go to Manage Org -> People
3. Open Network Tab
4. Type "hello" in the first name or any other text field

**Expected behavior**
The search functionality should be denounced, i.e. wait for the user to finish typing before sending a request to the  backend

**Actual behavior**
The backend is called immediately when input is changed, in this case, it is being called 5 times when typing "hello"

**Screenshots**

https://user-images.githubusercontent.com/28570857/232910861-1d988293-1eb2-417f-9fba-41e26e812d27.mov

**Additional details**
NA

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by prathamesh-mutkure**
*What kind of change does this PR introduce?*

Bug Fix

*Issue Number:*

Fixes #893

*Did you add tests for your changes?*

Yes, fixed tests

*Snapshots/Videos:*

https://user-images.githubusercontent.com/28570857/233140304-56356b48-e1a0-4819-a22e-8b2fa2d3dc70.mov

*If relevant, did you update the documentation?*

Not Relevant

*Summary*

- The debounce functionality didn't work
- I've fixed it
- Also fixed all the warnings for mocked data in tests

*Does this PR introduce a breaking change?*

No

*Other information*

NA

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by palisadoes** [comment]
Please fix the failing tests. The test coverage has declined with your changes.

**Comment by prathamesh-mutkure** [comment]
Hi @noman2002 @Rumone @Tajcore @ttaylor92 can you please review this

**Comment by Kevoniat** [comment]
@prathamesh-mutkure  why did the test coverage drop?

**Comment by palisadoes** [comment]
It's a rounding error. We've been seeing this a lot. A Drop of 0.01% is 1/10,000th.

Please review

File: src/screens/OrganizationPeople/OrganizationPeople.tsx [hunk 1]
```diff
@@ -84,6 +82,11 @@ function OrganizationPeople(): JSX.Element {
   })[1];
 
   useEffect(() => {
+    const filterData = {
+      firstName_contains: firstNameRef.current?.value ?? '',
+      lastName_contains: lastNameRef.current?.value ?? '',
+    };
+
     if (state === 0) {
```
Comment (Rumone) [inline src/screens/OrganizationPeople/OrganizationPeople.tsx:90]: Could you move the logic that changes the filter object and executes the refetch to another object. I see the same logic is used elsewhere in the screen.

**Review by Rumone — CHANGES_REQUESTED** [review]

File: src/screens/OrganizationPeople/OrganizationPeople.tsx [hunk 2]
```diff
@@ -106,12 +109,17 @@ function OrganizationPeople(): JSX.Element {
   if (memberError || usersError || adminError) {
     const error = memberError ?? usersError ?? adminError;
 
-    console.log(error);
+    console.error(error);
```
[the remark this answers is not in the document]
Comment (prathamesh-mutkure) [reply src/screens/OrganizationPeople/OrganizationPeople.tsx:112]: Done

File: src/screens/OrganizationPeople/OrganizationPeople.tsx (hunk 1 above)
Comment (prathamesh-mutkure) [reply src/screens/OrganizationPeople/OrganizationPeople.tsx:90]: Thanks for pointing that out, I've fixed that

**Comment by prathamesh-mutkure** [comment]
Hi @Rumone, I've fixed the issues, can you approve this PR?

**Comment by palisadoes** [comment]
Closing due to inactivity

**Review by palisadoes — APPROVED** [review]

**Pull request closed without merging**